azurelinuxagent/common/rdma.py

```diff
@@ -110,6 +109,26 @@ def get_rdma_version():
             logger.error(error_msg % driver_info_source)
             return
 
+    @staticmethod
+    def is_kvp_daemon_running():
+        """Look for kvp daemon names in ps -ef output and return True/False
+        """
+        # for centos, the hypervkvpd and the hv_kvp_daemon both are ok.
+        # for suse, it uses hv_kvp_daemon
+        kvp_daemon_names = ['hypervkvpd', 'hv_kvp_daemon']
+
+        exitcode, ps_out  = shellutil.run_get_output("ps -ef")
+        if exitcode != 0:
+            raise Exception('RDMA: ps -ef failed: %s' % ps_out)
+        for n in  kvp_daemon_names:
+            if n in ps_out:
+                logger.info('RDMA: kvp daemon (%s) is running' % n)
+                return True
+            else:
+                logger.verbose('RDMA: kvp daemon (%s) is not running' % n)
+        return False
+
```

Comment:
Fair points, I just ported the code do the same thing from waagent v2.0. It pretty much sounds like in the long term we need to find a better strategy to do RDMA, so I'm trying not to change many things and get back to v2.0 logic as close as possible.